bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/eGateHandler.java

```diff
@@ -0,0 +1,344 @@
+        Runnable runnable = new Runnable() {
+            @SuppressWarnings("null")
+            @Override
+            public void run() {
+                try {
+                    if (reader == null) {
+                        reader = new BufferedReader(new InputStreamReader(egateSocket.getInputStream()));
+                    }
+                    logger.debug("Socket State: {} to: {}", egateSocket.isConnected(), egateSocket.toString());
+                    if (reader.ready()) {
+                        String input = reader.readLine();
+                        logger.debug("Reader got from EGATE: {}", input);
+                        onData(input);
+                    }
+                } catch (IOException e) {
+                    logger.error("Error while reading command from Dominoswiss eGate Server {} ", e.toString());
+                    updateStatus(ThingStatus.OFFLINE);
+                }
+            }
+        };
+        refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, 1, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Finds and returns a child thing for a given UID of this bridge.
+     *
+     * @param uid uid of the child thing
+     * @return child thing with the given uid or null if thing was not found
+     */
+    public @Nullable Thing getThingByUID(ThingUID uid) {
+        Bridge bridge = getThing();
+
+        List<Thing> things = bridge.getThings();
+
+        for (Thing thing : things) {
+            if (thing.getUID().equals(uid)) {
+                return thing;
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    protected void onData(String input) {
+
+        // Instruction=2;ID=19;Command=1;Value=0;Priority=0;
+
+        Map<String, String> map = new HashMap<String, String>();
+        Thing blind;
+        // split on ;
+        String[] parts = input.split(";");
+
+        for (int i = 0; i < parts.length; i += 2) {
+            map.put(parts[i], parts[i + 1]);
+        }
+
+        // only use FSSReceive Commands
+        if (map.get("Instruction").toString() == "2") {
+            String id = map.get("ID");
+            ThingUID uid = registeredBlinds.get(map.get(id));
+            if (uid != null) {
+                blind = getThingByUID(uid);
+
+                List<Channel> channels = blind.getChannels();
+                logger.debug("Channels: {} of Blind {}", channels.toString(), blind.getConfiguration());
+
+                switch (map.get("Command")) {
+                    case "1":
+                        break;
+
+                    case "2":
+                        break;
+
+                    case "3":
+                        break;
+
+                    case "4":
+                        break;
+                    default:
+                        break;
+                }
+            } else {
+                logger.error("Error no blind registered in eGate Bridge. Please add some blinds");
```

Comment:
This should not be an ERROR level, see logging levels in the docs and I am not sure there is value in logging this at all but leave this up to you since I don't know the protocol and when this code will get triggered.